deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/KerasModelEndToEndTest.java

```diff
@@ -0,0 +1,156 @@
+package org.deeplearning4j.nn.modelimport.keras;
+
+import lombok.extern.slf4j.Slf4j;
+import org.deeplearning4j.eval.ROCMultiClass;
+import org.deeplearning4j.nn.multilayer.MultiLayerNetwork;
+import org.junit.Test;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.io.ClassPathResource;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.*;
+
+import static org.junit.Assert.*;
+
+/**
+ * Unit tests for end-to-end Keras model import.
+ *
+ * TODO: find a more elegant fix for the per-layer activation comparisons
+ * since some layers (e.g., ActivationLayer) overwrite previous
+ * layer's activations when run in train=false mode.
+ *
+ * TODO: ndarray.equalsWithEps(ndarray) appears to be broken
```

Comment:
Not sure whether it's user error or a bug in ND4J. I just know that the stored Keras and computed DL4J activations are similar within 6-7 digits of precision, but `equalsWithEps` returns false. Investigating.